src/queries/puzzle.py

```diff
@@ -128,3 +128,26 @@ async def delete_puzzle(puzzle_id: str):
     await aconn.close()
 
     return True
+
+
+async def get_leaderboard() -> tuple[str, int]:
+    aconn = await psycopg.AsyncConnection.connect(DATABASE_URL)
+    acur = aconn.cursor()
+
+    await acur.execute(
+        """
+        SELECT t.team_name, t.puzzle_solved
+        FROM public.teams AS t LEFT JOIN public.submissions AS s
+        ON (t.team_name = s.team_name)
+        AND s.submission_is_correct = TRUE
+        GROUP BY t.team_name
+        ORDER BY t.puzzle_solved DESC, MAX(s.submission_time) ASC
+        """
```

Comment:
in the case that the puzzle_solved and submission_time is the same, we should sort on team_name as well (only ever likely for when teams have 0 solves)

Reply:
oh yeah maybe. I didn't think it mattered for the order of teams with no solves but I could do that.